[E]69.cpp

```diff
@@ -0,0 +1,19 @@
+// Problem: 69. Sqrt(x) - Easy
+// Link: https://leetcode.com/problems/sqrtx/
+
+class Solution {
+public:
+    int mySqrt(int x) {
+        // cout << (int)sqrt(INT_MAX) << endl;
+        // I must cout to find the max of range first, before searching.
+        int l = 0, r = 46340;
```

Comment:
Cái r nên để là x + 1, thay vì 1 số fix.

Reply:
Như vậy nếu x = INT_MAX thì (r = x + 1) sẽ tràn ngoài giới hạn của integer.
Ngoài ra, anh dùng r trong trường hợp này là căn của giới hạn trên, nên có vẻ đặt bằng x là chưa hợp lý lắm. Chú giải thích ý tưởng của chú được ko?

Reply:
Với bài này thì có thể đặt r = x duoc. Khi đó thì cái lệnh mid = l + (r - l) / 2; bên dưới sẽ giúp mình tránh tràn số.

Reply:
Tuy nhiên khi đó thì thuật toán của anh sẽ nảy sinh vấn đề, đó là (r * r) sẽ bị tràn số.

Reply:
Uh vậy thì tốt nhất vẫn nên để l,r là long.

Reply:
Okay. Sau khi đọc discussion anh mới phát hiện ra là: giới hạn bên trái "l", là không cần thiết vì nếu đặt r = x, thì chỉ cần giảm r là sẽ tới giá trị cần tìm.

Reply:
Cũng có thể nhưng mà giờ a cần chú code thật chuẩn binary search đã 👍

Reply:
Ok. Wait for next version please. Cái giới hạn trái kia áp dụng thuật toán Newton, có lẽ ko nên dùng :))

Reply:
👍